docs/adr/adr-004-state-interaction.md

```diff
@@ -0,0 +1,121 @@
+# ADR #004: Celestia-Node State Interaction for March 2022 Testnet
+
+<hr style="border:3px solid gray"> </hr>
+
+## Authors
+
+@renaynay
+
+## Changelog
+
+* 2022-01-14: initial draft
+
+<hr style="border:2px solid gray"> </hr>
+
+## Context
+
+Currently, celestia-node lacks a way for users to submit a transaction to celestia-core as well as access its own state
+and public state of others.
+
+## Decision
+
+Both celestia **light** and **full** nodes will run `StateService`. 
+`StateService` will be responsible for managing the RPC endpoints provided by `StateAccessor`.
+
+`StateAccessor` will be an interface for interacting with celestia-core in order to retrieve account-related information
+as well as submit transactions.
+
+### `StateService`
+
+`StateService` will expose several higher-level RPC endpoints for users to access the methods provided by the 
+`StateAccessor`.
+
+```go
+type StateService struct {
+    accessor StateAccessor
+}
+``` 
+
+### `StateAccessor`
+
+`StateAccessor` is defined by the two methods listed below and may be expanded in the future to include more methods 
+related to accounts and transactions in later iterations.
+
+```go
+type StateAccessor interface {
+    func AccountBalance(context.Context, Account) (Balance, error)
+    func SubmitTx(context.Context, Tx) error
```

Comment:
It is a bit unclear which type `Tx` refers to here. In the core impl it says `where Tx implements sdk.Msg`. Is that generally the assumption in the iface here too?

Reply:
Yes. I can document it in the interface docs